helix-core/src/main/java/org/apache/helix/controller/rebalancer/condition/RebalanceCondition.java

```diff
@@ -0,0 +1,5 @@
+package org.apache.helix.controller.rebalancer.condition;
+
+public interface RebalanceCondition {
+  boolean shouldPerformRebalance();
```

Comment:
I think we need a generic string -> string map for input ? Or directly pass the cache object.

Because condition is a piece of code logic but the real computation is happening at runtime with input.